test/framework/src/main/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -165,6 +174,106 @@ public void testNodeAttributes() throws IOException {
         }
     }
 
+    public void testNodeConstructionWithDefaultPathDataSet() throws IOException {
+        final Path home = createTempDir().toAbsolutePath();
+        final Path zero = createTempDir().toAbsolutePath();
+        final Path one = createTempDir().toAbsolutePath();
+        final Path defaultPathData = createTempDir().toAbsolutePath();
+        final Settings settings = Settings.builder()
+                .put("path.home", home)
+                .put("path.data.0", zero)
+                .put("path.data.1", one)
+                .put("default.path.data", defaultPathData)
+                .put("http.enabled", false)
+                .put("transport.type", "mock-socket-network")
+                .build();
+        Files.createDirectories(defaultPathData.resolve("nodes/0"));
+        final boolean indexExists = randomBoolean();
+        if (indexExists) {
+            final int numberOfIndices = randomIntBetween(1, 3);
+            for (final String index :
+                    IntStream.range(0, numberOfIndices).mapToObj(i -> UUIDs.randomBase64UUID()).collect(Collectors.toList())) {
+                Files.createDirectories(defaultPathData.resolve("nodes/0/indices").resolve(index));
+            }
+        }
+        Supplier<MockNode> constructor = () -> new MockNode(settings, Collections.singletonList(MockTcpTransportPlugin.class));
```

Comment:
I don't think they should be combined. The first block is test setup, the second block is test execution.